.github/workflows/release.yml

```diff
@@ -11,79 +11,153 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        arch: [auto, aarch64]
+        arch: [native, aarch64, universal2]
         py: [cp37, cp38, cp39, cp310]
-        include:
-          - os: macos-latest
-            py: cp38
-            arch: universal2
-          - os: macos-latest
-            py: cp39
-            arch: universal2
-          - os: macos-latest
-            py: cp310
-            arch: universal2
         exclude:
           - os: windows-latest
             arch: aarch64
+          - os: windows-latest
+            arch: universal2
           - os: macos-latest
-            arch: aarch64
-    env:
-      CIBW_BUILD: ${{ matrix.py }}-*
-      CIBW_ARCHS_LINUX: ${{ matrix.arch }}
+            arch: aarch64 
+          - os: ubuntu-latest
+            arch: universal2
     steps:
-      - uses: actions/checkout@v2
+# Step for debugging    
+      - name: Show environment
+        run: env | grep ^GITHUB
+# Step for debugging
+      - name: Show CI env
+        run: env | grep ^CI  
+
+      - uses: actions/checkout@v3
         with:
           submodules: true
 
       - if: ${{ matrix.arch == 'aarch64' }}
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2 
+
+      - uses: awvwgk/setup-fortran@main
+        id: setup-fortran
+        with:
+          compiler: gcc
+          version: 11
+
+# Step for debugging
+      - run: ${{ env.FC }} --version
+        env:
+          FC: ${{ steps.setup-fortran.outputs.fc }}     
+
+      - if: ${{ matrix.os == 'windows-latest' }}
+        uses: pypa/cibuildwheel@v2.9.0
+        env:
+          CIBW_BUILD_VERBOSITY: 3
+          CIBW_BUILD: ${{ matrix.py }}-*
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_SKIP: "*musllinux*"
+          CMAKE_GENERATOR: "MinGW Makefiles"
+          FC: ${{ steps.setup-fortran.outputs.fc }}
+          CI: true
 
-      - uses: pypa/cibuildwheel@v2.3.1
+      - if: ${{ matrix.os != 'windows-latest' }}
+        uses: pypa/cibuildwheel@v2.9.0
         env:
+          # Check setuptools version
+          CIBW_BUILD_VERBOSITY: 3
           CIBW_BUILD: ${{ matrix.py }}-*
           CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_SKIP: "*musllinux*"
+          CIBW_TEST_SKIP: "*-macosx_universal2:arm64" 
+          CI: true  
```

Comment:
This is already set by GitHub Actions (and all other CI providers that I know of).

Reply:
`CI: true` is not needed anymore. I set it because tests required CI variable, but it was not found on Linux. I think it is because of `manylinux`.

Reply:
If you need outer environment variables, you can use:

```toml
[tool.cibuildwheel]
linux.environment-pass = ["CI"]
```

Though honestly, we should probably pass this one by default.

Reply:
I will keep that in mind.